src/pykeen/nn/modules.py

```diff
@@ -1421,8 +1461,19 @@ class QuatEInteraction(
     .. seealso:: :func:`pykeen.nn.functional.quat_e_interaction`
     """
 
+    # with k=4
+    entity_shape: Sequence[str] = ("dk",)
+    relation_shape: Sequence[str] = ("dk",)
     func = pkf.quat_e_interaction
 
+    def __init__(self) -> None:
+        """Initialize the interaction module."""
+        super().__init__()
+        self.register_buffer(name="table", tensor=quaterion_multiplication_table())
```

Comment:
does this automatically put the tensor returned by `quaterion_multiplication_table()` on to the appropriate device?

Reply:
Yes. buffers and parameters are (recursively) moved along with the module.